test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,851 @@
+			// According to Restricted TM policy: only allow allocations from the minimum number of NUMA nodes.
+			// Look at each resource request, see what the minimum number of NUMA nodes are required to
+			// satisfy that resource request. Allow alignment to that number of NUMA nodes for all resources.
+			// Hence the pod should fail with TopologyAffinityError
+			err := checkPodEvent(testPod, "TopologyAffinityError")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+			err = mm1.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove test pod")
+		})
+
+		It("[test_id:60694] Accept guaranteed pod requesting resources that can be satisfied by 2 numa nodes together", func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			// no. of hugepages is 20 * 2 (numazones). 40Mi
+			// we are asking for 30Mi, so it needs 2 numazones combined to
+			// satisfy the requirement
+			mm2.noOfhpgs = "30Mi"
+			testPod := mm2.createPodTemplate(profile, true, targetNode)
+			// Initialize test pod, check if the pod uses both numa node  0 and 1
+			err := initializePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "unable to initialize Pod")
+			numaZone, err := GetMemoryNodes(testPod, targetNode)
+			// Expect both numa nodes to be used by pod
+			Expect(numaZone).To(Equal("0-1"))
+			Expect(err).ToNot(HaveOccurred(), "Pod's numa affinity is %s instead of %s", numaZone, "0-1")
+			err = mm2.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove test pod")
+		})
+
+		It("[test_id:60695] Allow burstable pod with hugepages", func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			mm2.noOfhpgs = "8Mi"
+			testPod := mm2.createPodTemplate(profile, false, targetNode)
+			// Initialize test pod, check if the pod uses both numa node  0 and 1
+			err := initializePod(testPod) // "0-1", targetNode)
+			Expect(err).ToNot(HaveOccurred(), "Unable to initialize pod")
+			numaZone, err := GetMemoryNodes(testPod, targetNode)
+			// Expect both numa nodes to be used by pod
+			Expect(numaZone).To(Equal("0-1"), "Pod's numa affinity is %s instead of %s", numaZone, "0-1")
+			err = mm2.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove test pod")
+		})
+
+		AfterAll(func() {
+			By("Reverting the Profile")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			currentSpec, _ := json.Marshal(profile.Spec)
+			spec, _ := json.Marshal(initialProfile.Spec)
+			// revert only if the profile changes.
+			if !bytes.Equal(currentSpec, spec) {
```

Comment:
maybe it's better to use more high level compare like: `equality.Semantic.DeepEqual`
because byte comparison might be false if fields are not sort at the same order.